src/web-component.js

```diff
@@ -34,13 +35,22 @@ class WebComponent extends HTMLElement {
   }
 
   static get observedAttributes() {
-    return ["code", "sense_hat_always_enabled", "step_number"];
+    return ["code", "sense_hat_always_enabled", "step_number", "steps"];
   }
 
   attributeChangedCallback(name, _oldVal, newVal) {
-    // console.log(name, newVal)
-    this.componentAttributes[name] = newVal;
-
+    let value;
+
+    if (["sense_hat_always_enabled"].includes(name)) {
+      value = Boolean(newVal);
+    } else if (["step_number"].includes(name)) {
+      value = parseInt(newVal);
+    } else if (["steps"].includes(name)) {
+      value = JSON.parse(newVal);
+    } else {
+      value = newVal;
+    }
```

Comment:
Just curious but is there a reason to check whether an array includes the name rather than `if name == "steps"`?

This could also be turned into a switch statement (although I appreciate that this might be a personal preference of mine!)

Reply:
Yes, the main reason is so it can be extended when we add more attributes of different types